envs/quadrotors/quad_multi_model.py

```diff
@@ -24,13 +117,30 @@ def __init__(self, cfg, obs_space, timing, self_obs_dim=18, obstacle_obs_dim=6,
             self.neighbor_obs_dim = 9  # include goal pos info
         elif self.neighbor_obs_type == 'pos_vel':
             self.neighbor_obs_dim = 6
+        elif self.neighbor_obs_type == 'attn':
+            self.neighbor_obs_dim = 11
         elif cfg.neighbor_obs_type == 'none':
             # override these params so that neighbor encoder is a no-op during inference
             self.neighbor_obs_dim = 0
             self.num_use_neighbor_obs = 0
         else:
             raise NotImplementedError(f'Unknown value {cfg.neighbor_obs_type} passed to --neighbor_obs_type')
 
+        # encode the neighboring drone's observations
+        neighbor_encoder_out_size = 0
+        if self.num_use_neighbor_obs > 0:
+            if self.neighbor_obs_type == 'pos_vel' or self.neighbor_obs_type == 'pos_vel_goals':
+                self.neighbor_encoder = QuadNeighborhoodEncoderDeepsets(cfg, self.neighbor_obs_dim,
+                                                                        self.neighbor_hidden_size, self.use_spectral_norm,
+                                                                        self.self_obs_dim, self.num_use_neighbor_obs)
+            elif self.neighbor_obs_type == 'attn':
```

Comment:
These have to be orthogonal settings, the obs types and the type of neighbor encoder.
We need to run ablation deepsets vs attn